data/transactions/logic/eval.go

```diff
@@ -296,7 +478,12 @@ func Check(program []byte, params EvalParams) (cost int, err error) {
 	cx.EvalParams = params
 	cx.program = program
 	for (cx.err == nil) && (cx.pc < len(cx.program)) {
+		prevpc := cx.pc
 		cost += cx.checkStep()
+		if cx.pc == prevpc {
```

Comment:
since we don't have backward jump, wouldn't it be a stronger predicate to use `cx.pc <= prevpc` ?

Reply:
pushed pavel's change from https://github.com/justicz/go-algorand/pull/110